### ISSUE TITLE
Support for a "pulls" query

### ISSUE THREAD

**Issue by BryanLa** [sshaw/export-pull-requests#14]
Any chance you're looking at adding support for a query to the /pulls endpoint, instead of /issues? 

It would be great to be able to query for only PRs, and use other switches for [/pulls query string parameters](https://developer.github.com/v3/pulls/#parameters) such as "base" and "head".

**Comment by sshaw** [comment]
Hi, thanks for the info. Sounds like a good idea.

Currently there's an `--export` (`-x`) option. Given what you've proposed, it sounds like when one specifies `--export pr` we should use the `/pulls` endpoint and otherwise do what's currently being done.

Do you know if the options currently being used (assignee, milestone, etc.. ) when `/issues`  is queried are supported by `/pulls` (yes I know you linked to the docs –thank you!– but hoping you can assist me here 😉) and are the parameter names the same across the two? 

Also, any ideas if Bitbucket or GitLab have similar options?

**Comment by BryanLa** [comment]
Thanks @sshaw! 

Yes, I think that would be a great idea to alter the API endpoint, based on the `export` switch value.

As far as the options for each GitHub API endpoint, it looks like the only ones shared across the two are `state`, `sort` and `direction`. [Here's the reference](https://developer.github.com/v3/issues/#parameters-1) for `/issues`. Another thought... what if you provided a freeform `-q`/`--query-string` switch, and allow folks to plug in their own, and pass them in the API query?

Sorry, I don't know about Bitbucket and GitLab... maybe the above could be enabled just for `-p=github`, initially?

**Comment by sshaw** [comment]
> what if you provided a freeform -q/--query-string switch, and allow folks to plug in their own, and pass them in the API query?

Yeah seems like the best direction to go. But maybe have specific switches to normalize and/or simplify common options across services. 

Maybe something like  `-q foo=bar -q baz=123`. Will look into this...

**Comment by sshaw** [comment]
`-q 'foo=bar&baz=123'` seems cumbersome and since in most (all?) *nix shells `&` puts a job in the background specifying multiple `-q`s seems more user-friendly. Plus code would have to parse whereas the other would be parsed by option processing lib (I _think!).
